binrz/rz-test/load.c

```diff
@@ -125,152 +125,153 @@ RZ_API RzPVector *rz_test_load_cmd_test_file(const char *file) {
 			if (!test->cmds.value) {
-				eprintf (LINEFMT "Error: Test without CMDS key\n", file, linenum);
+				eprintf(LINEFMT "Error: Test without CMDS key\n", file, linenum);
 				goto fail;
 			}
 			if (!(test->expect.value || test->expect_err.value)) {
-				eprintf (LINEFMT "Error: Test without EXPECT or EXPECT_ERR key"
-				         " (did you forget an EOF?)\n", file, linenum);
+				eprintf(LINEFMT "Error: Test without EXPECT or EXPECT_ERR key"
+						" (did you forget an EOF?)\n",
+					file, linenum);
 				goto fail;
 			}
-			rz_pvector_push (ret, test);
-			test = rz_test_cmd_test_new ();
+			rz_pvector_push(ret, test);
+			test = rz_test_cmd_test_new();
 			if (!test) {
 				goto beach;
 			}
 			continue;
 		}
 
-#define DO_KEY_STR(key, field) \
-		if (strcmp (line, key) == 0) { \
-			if (test->field.value) { \
-				free (test->field.value); \
-				eprintf (LINEFMT "Warning: Duplicate key \"%s\"\n", file, linenum, key); \
-			} \
-			if (!val) { \
-				eprintf (LINEFMT "Error: No value for key \"%s\"\n", file, linenum, key); \
-				goto fail; \
-			} \
-			test->field.line_begin = linenum; \
-			test->field.value = read_string_val (&nextline, val, &linenum); \
-			test->field.line_end = linenum + 1; \
-			if (!test->field.value) { \
-				eprintf (LINEFMT "Error: Failed to read value for key \"%s\"\n", file, linenum, key); \
-				goto fail; \
-			} \
-			continue; \
-		}
-
-#define DO_KEY_BOOL(key, field) \
-		if (strcmp (line, key) == 0) { \
-			if (test->field.value) { \
-				eprintf (LINEFMT "Warning: Duplicate key \"%s\"\n", file, linenum, key); \
-			} \
-			test->field.set = true; \
-			/* Strip comment */ \
-			char *cmt = strchr (val, '#'); \
-			if (cmt) { \
-				*cmt = '\0'; \
-				cmt--; \
-				while (cmt > val && *cmt == ' ') { \
-					*cmt = '\0'; \
-					cmt--; \
-				} \
-			} \
-			if (!strcmp (val, "1")) { \
-				test->field.value = true; \
-			} else if (!strcmp (val, "0")) { \
-				test->field.value = false; \
-			} else { \
-				eprintf (LINEFMT "Error: Invalid value \"%s\" for boolean key \"%s\", only \"1\" or \"0\" allowed.\n", file, linenum, val, key); \
-				goto fail; \
-			} \
-			continue; \
-		}
-
-#define DO_KEY_NUM(key, field) \
-		if (strcmp (line, key) == 0) { \
-			if (test->field.value) { \
-				eprintf (LINEFMT "Warning: Duplicate key \"%s\"\n", file, linenum, key); \
-			} \
-			test->field.set = true; \
-			/* Strip comment */ \
-			char *cmt = strchr (val, '#'); \
-			if (cmt) { \
-				*cmt = '\0'; \
-				cmt--; \
-				while (cmt > val && *cmt == ' ') { \
-					*cmt = '\0'; \
-					cmt--; \
-				} \
-			} \
-			char *endval; \
-			test->field.value = strtol (val, &endval, 0); \
-			if (!endval || *endval) { \
-				eprintf (LINEFMT "Error: Invalid value \"%s\" for numeric key \"%s\", only numbers allowed.\n", file, linenum, val, key); \
-				goto fail; \
-			} \
-			continue; \
-		}
+#define DO_KEY_STR(key, field)                                                                               \
```

Comment:
Is there an option to disable this? I think it's quite annoying that every time you fix one line in a macro you also have to change every other line in that macro (it also "hides" the git history for no good reason, imho.

Reply:
I think we just should fence off the big multiline macroses. `clang-format` doesn't handle them well. We can report it as a bug.

Reply:
Disabled.